contracts/contracts/ocr2/SequencerUptimeFeed/library.cairo

```diff
@@ -0,0 +1,258 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin, SignatureBuiltin
+from starkware.starknet.common.syscalls import (
+    get_tx_info,
+    get_block_timestamp,
+    get_caller_address,
+    get_block_number,
+)
+from starkware.cairo.common.math import assert_not_zero, assert_le
+from starkware.cairo.common.math_cmp import is_le
+from starkware.cairo.common.bool import TRUE, FALSE
+
+from utils import assert_boolean
+from ocr2.interfaces.IAggregator import Round, AnswerUpdated, NewRound
+from SimpleReadAccessController.library import simple_read_access_controller
+from ownable import Ownable_only_owner
+
+@event
+func RoundUpdated(status : felt, transmission_timestamp : felt):
+end
+
+@event
+func UpdateIgnored(
+    latest_status : felt, latest_timestamp : felt, incoming_status : felt, incoming_timestamp : felt
+):
+end
+
+@event
+func L1SenderTransferred(prev : felt, cur : felt):
+end
+
+@storage_var
+func s_l1_sender() -> (address : felt):
+end
+
+@storage_var
+func s_l2_cross_domain_messenger() -> (address : felt):
+end
+
+@storage_var
+func s_rounds(id : felt, field : felt) -> (res : felt):
+end
+
+@storage_var
+func s_latest_round_id() -> (res : felt):
+end
+
+func require_l1_sender{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    address : felt
+):
+    let (l1_sender) = s_l1_sender.read()
+    with_attr error_message("invalid sender"):
+        assert l1_sender = address
+    end
+
+    return ()
+end
+
+func require_valid_round_id{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    round_id : felt
+):
+    let (lateset_round_id) = s_latest_round_id.read()
+
+    with_attr error_message("invalid round_id"):
+        assert_not_zero(round_id)
+        # TODO: do we need to check if uint80 is overflown?
```

Comment:
is this still a todo?

Reply:
`Solidity` implementation uses `uint80` and as I remember ensures that it doesn't overflow. I'm not sure how critical that's why a left this todo. So answer would depend on your requirements :)

Reply:
Discussed with @archseer

Reply:
@archseer please resolve if ok?